dask_bigquery/tests/test_core.py

```diff
@@ -0,0 +1,100 @@
+import random
+
+import pandas as pd
+import pytest
+from distributed.utils_test import cluster_fixture  # noqa: F401
+from distributed.utils_test import client, loop  # noqa: F401
+from google.cloud import bigquery
+
+from dask_bigquery import read_gbq
+
+# These tests are run locally and assume the user is already athenticated.
+# It also assumes that the user has created a project called dask-bigquery.
+
+
+def gen_data(size=10):
+    records = [
+        {
+            "name": random.choice(["fred", "wilma", "barney", "betty"]),
+            "number": random.randint(0, 100),
+            "idx": i,
+        }
+        for i in range(size)
+    ]
+    return pd.DataFrame(records)
+
+
+def push_data():
+    "Push data to BigQuery using pandas gbq"
+    df = gen_data()
+
+    pd.DataFrame.to_gbq(
+        df,
+        destination_table="dataset_test.table_test",
+        project_id="dask-bigquery",
+        chunksize=5,
+        if_exists="append",
+    )
+
+    return df
+
+
+# test simple read
+def test_read_gbq(client):
+    """Test simple read of data pushed to BigQuery using pandas-gbq"""
+    try:
+        # delete data set if exists
+        bq_client = bigquery.Client()
+        bq_client.delete_dataset(
+            dataset="dask-bigquery.dataset_test",
+            delete_contents=True,
+        )
+        bq_client.close()
+    except:  # if data doesn't exisit continue is that Value Error?
+        pass
+    # create data
+    df = push_data()
+
```

Comment:
This seems like the kind of stuff that might be good as a fixture.  

```python
@pytest.fixture
def dataset():
    bq = bigquery.Client()
    ...
    push_data()
    yield dask-bigquery.dataset_test
    bq.delete_dataset(...)
```

```python
def test_read_gbq(client, dataset):
    project_id, dataset_id = dataset.split(".")
    ddf = read_gbq(dataset_id=dataset_id, project_id=project_id, ...)
```

Reply:
I'll take a look and see if we can use fixtures, but maybe not with the delete part. The purpose of the try/except here is to delete the data set if it exists before pushing the data. If I delete the dataset right after pushing it then I won't be able to read it,
```python
@pytest.fixture
def dataset():
    bq = bigquery.Client()
    ...
    push_data()
    yield dask-bigquery.dataset_test
    bq.delete_dataset(...)
```
Maybe I'm missing something, but from this snippet I will be yielding the name of location of the dataset and then deleting it, by the time I get to `read_gbq` the data would be removed, and there would be nothing to read. 
That being said, there might be a way of still using a fixture of some form.

Reply:
The test runs during the yield.  So this test says "set up a dataset, then give the path of that dataset to the test, let the test do its thing,  and then once the test is done clean up the dataset"

If you're concerned that the dataset might exist beforehand then you could add it to the part of the fixture before the yield as well.